### PR TITLE
Add kdestroy -p option

### PR DESCRIPTION
**Pull request by greghudson**
[Requested by simo.]

Add an option to destroy a cache within a collection by principal
name.  This option can be used together with -c to specify the
collection.  Also document that kdestroy -A and -c can be used
together (ticket 8602).

### CODE REVIEW

**Comment by simo5** [comment]
Looks great!
Thanks!

**Comment by greghudson** [comment]
I meant to call out that as implemented in the PR, kdestroy -p suppresses the "Other credential caches present" warning, on the theory that if you're searching for a specific client principal you probably know that already.

File: src/clients/kdestroy/kdestroy.c [hunk 1]
```diff
@@ -122,6 +134,11 @@ main(int argc, char *argv[])
         }
     }
 
+    if (all && princ_name != NULL) {
+        fprintf(stderr, _("-a option is exclusive with -p option\n"));
```
Comment (frozencemetery) [inline src/clients/kdestroy/kdestroy.c]: Capitalized - `-A`

Comment (greghudson) [reply src/clients/kdestroy/kdestroy.c]: Fixed, thanks.

**Pull request merged**